langkit/compile_context.py

```diff
@@ -1703,23 +1719,34 @@ def emit(self, lib_root, check_only=False, warnings=None,
         # Compute the list of passes to run:
 
         # First compile the DSL
-        all_passes = self.compilation_passes
+        self.all_passes = self.compilation_passes
 
         # Then, if requested, emit code for the generated library
-        if not check_only:
-            all_passes.append(
+        if not self.check_only:
+            self.all_passes.append(
                 self.prepare_code_emission_pass(lib_root, **kwargs))
 
-            all_passes.extend(self.code_emission_passes(annotate_fields_types))
+            self.all_passes.extend(self.code_emission_passes())
 
             # Run plugin passes at the end of the pipeline
-            all_passes.extend(plugin_passes)
+            self.all_passes.extend(plugin_passes)
+
+        for p in self.all_passes:
+            if p.is_optional and p.name in explicit_passes_triggers.keys():
```

Comment:
I did not realize this until I saw the testcase: having to pass `--pass-on "foo bar"` (i.e. having spaces in pass names) will be cumbersome to use. What about introducing the concept of “command-line name” in pass classes, for instance replacing spaces with dashes and converting to lowercase?

Reply:
I'm OK to do this in a subsequent PR